soroban-env-host/src/e2e_invoke.rs

```diff
@@ -151,8 +160,15 @@ pub fn get_ledger_changes(
                 new_live_until_ledger: 0,
             });
         }
-        if init_storage_snapshot.has(key)? {
-            let (old_entry, old_live_until_ledger) = init_storage_snapshot.get(key)?;
+        #[cfg(any(test, feature = "unstable-next-api"))]
+        let entry_with_live_until = init_storage_snapshot.get(key)?;
+        #[cfg(not(any(test, feature = "unstable-next-api")))]
+        let entry_with_live_until = if init_storage_snapshot.has(key)? {
```

Comment:
The fact that `test` is included in here as an any condition suggests that the non-unstable code is not tested. How can we still test it?

Reply:
I don't think there is anything to test in this particular case. `SnapshotSource` is not really covered directly. The new field is covered in the tests as well and I don't see much value in covering not returning this field.